src/uct/api/uct.h

```diff
@@ -1329,6 +1390,29 @@ ucs_status_t uct_md_config_read(const char *name, const char *env_prefix,
                                 uct_md_config_t **config_p);
 
 
+
+/**
+ * @ingroup UCT_MD
+ * @brief Check if remote sock address is reachable from the memory domain.
+ *
+ * This function checks if a remote sock address can be reached from a local
+ * memory domain. Reachability can be checked in local or remote mode.
+ *
+ * @param [in]  md         Memory domain to check reachability from.
+ * @param [in]  sockaddr   Socket address to check reachability to.
+ * @param [in]  mode       Mode for checking reachability, as defined in @ref
+ *                         uct_sockaddr_reachability_t.
+ *                         Indicates if reachability is tested on the server side -
+ *                         for binding to the given sockaddr, or on the
+ *                         client side - for connecting to the given remote
+ *                         peer's sockaddr.
+ *
+ * @return Nonzero if reachable, 0 if unreachable.
+ */
+int uct_md_is_sockaddr_reachable(uct_md_h md, const ucs_sock_addr_t *sockaddr,
+                                 uct_sockaddr_reachability_t mode);
+
+
```

Comment:
Sorry for not replying sooner...my system was stuck in an upgrade and was "unreachable" for some time :(
I'm pretty much okay with the term 'reachable'.  However, maybe 'accessible' is better?  I'm interpreting the use of 'reachable' here as in network reachable and not necessarily that one can connect to the given sockaddr.